readthedocs/doc_builder/python_environments.py

```diff
@@ -344,6 +344,8 @@ def install_core_requirements(self):
                     negative='readthedocs-sphinx-ext<2.2',
                 ),
             ])
+            if self.project.has_feature(Feature.USE_SPHINX_LATEST):
+                requirements.extend(["jinja2<3.1.0"])
```

Comment:
If Jinja 3.1 only affects Sphinx 1.8, this logic looks inverted -- it's positive for projects that are using the latest version of Sphinx